src/NodeChanger/IdentifierRenamer.php

```diff
@@ -33,10 +40,15 @@ public function renameNodeWithMap(Node $node, array $renameMethodMap): void
 
     private function ensureNodeHasIdentifier(Node $node): void
     {
-        if (! in_array(get_class($node), [
-            ClassConstFetch::class, MethodCall::class, PropertyFetch::class, StaticCall::class,
-        ])) {
-            throw new Exception('The given Node does not have a valid Identifier.');
+        if (in_array(get_class($node), $this->nodeClassesWithIdentifier, true)) {
+            return;
         }
+
+        throw new NodeMissingIdentifierException(sprintf(
+            'Node "%s" does not contain a "$name" property with "%s". Pass only one of "%s".',
+            get_class($node),
+            Identifier::class,
+            implode('", "', $this->nodeClassesWithIdentifier)
+        ));
```

Comment:
:+1: For custom Exception

Reply:
Thanks for inspiration!

Reply:
Why not invert the logic here?
```php
if (!in_array(get_class($node), $this->nodeClassesWithIdentifier, true)) {
     throw new NodeMissingIdentifierException(sprintf(
            'Node "%s" does not contain a "$name" property with "%s". Pass only one of "%s".',
            get_class($node),
            Identifier::class,
            implode('", "', $this->nodeClassesWithIdentifier)
))};
```
So we can avoid empty `return;`

Reply:
Why not invert the logic here? So we can avoid empty `return;`

Reply:
It's called *early return statement*. The goal is to resolve the easy part first.

Than resolve more difficult part.

So the complexity of call is related to complexity of algorithm. From easiest to most difficult.

Reply:
Here it's explained better: https://github.com/jupeter/clean-code-php#avoid-nesting-too-deeply-and-return-early-part-1

Reply:
_Early return_, I have heard about them. But in this case, we have `in_array` check. Just denying it, by prefix with `!`, isn't better? Isn't less or more complexity, just denying :thinking:

Reply:
Well, what if we need to add another check, which requires the `$name` property exist on the object?
(Which is the case now, because we call `$node->name' but rather implicit).

Reply:
Now I understood your point. Thanks for teach me :man_teacher: